### ISSUE TITLE
System.InvalidOperationException: "The stored extension value has a type of 'Google.Api.HttpRule,

### ISSUE THREAD

**Issue by SyntaxEvg** [protocolbuffers/protobuf#11628]
**Please help me understand the reason for this exception net7 VS 2022** 
System.InvalidOperationException: "The stored extension value has a type of 'Google.Api.HttpRule, service, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. This is a different from the requested type of 'Google.Api. HttpRule, Google.Api.CommonProtos, Version=2.5.0.0, Culture=neutral, PublicKeyToken=3ec5ea7f18953e47'."

**Comment by jskeet** [comment]
It sounds like you've *probably* generated the `HttpRule` class yourself, which I'd suggest you avoid doing - just use the one within Google.Api.CommonProtos. I won't be able to help any more than that until you've provided more information about how you ran into this problem though.

**Comment by SyntaxEvg** [comment]
thanks for the answer. I marked the code on the guide, you have the opportunity to run it yourself, at the stage of starting the assembly you will see an error
https://github.com/SyntaxEvg/WorkingSOAP_REST_gRPC/commit/dbcedd6af7a7a09a8a0a3508148c89bbea4f53e4

**Comment by jskeet** [comment]
But still no full stack trace, no explanation of what you're trying to do, etc. I'll have a look, and I'm pretty sure that it'll be exactly as I said: you've got copies of the various protos yourself, which you're generating, instead of using a reference to Google.Api.CommonProtos. I can't say I've ever used the MSBuild tools for Protobuf, and I don't know whether that'll make things tricky or not - I'll have a look in a couple of hours when I've started work properly, but *please* give more information in future questions.

**Comment by SyntaxEvg** [comment]
> But still no full stack trace, no explanation of what you're trying to do, etc. I'll have a look, and I'm pretty sure that it'll be exactly as I said: you've got copies of the various protos yourself, which you're generating, instead of using a reference to Google.Api.CommonProtos. I can't say I've ever used the MSBuild tools for Protobuf, and I don't know whether that'll make things tricky or not - I'll have a look in a couple of hours when I've started work properly, but _please_ give more information in future questions.

you are right, close the question, you need to do as you wrote so that it does not duplicate modules ...

**Comment by jskeet** [comment]
Closing as resolved - I did get as far as checking out the repo, but it looks like really any questions should be directed at the gRPC-Web team, given https://learn.microsoft.com/en-us/aspnet/core/grpc/grpcweb etc. I don't know enough about how projects are *expected* to be configured. Further issues like this are probably better reported in https://github.com/grpc/grpc-dotnet unless it's been isolated to be *really* about the protobuf runtime misbehaving. Good luck with your project.
